### ISSUE TITLE
Cannot save values to a profile

### ISSUE THREAD

**Issue by pjmonks** [MauroDataMapper/mdm-core#129]
**Describe the bug**
When sending a `POST /api/{domainType}/{id}/profile/{namespace}/{name}` request to save profile values, the response returned clears all values submitted and loses information. However, a `200 OK` status is returned suggesting something went right.

First noticed this when testing the [DOI profile](https://github.com/MauroDataMapper-Plugins/mdm-plugin-digital-object-identifiers), however it is not just related to that as I can replicate with another profile.

**To Reproduce**
From the UI:
1. Select any model that is editable.
2. In the "Description" tab, click the dropdown menu to select "Add new profile"
3. Enter any values into your chosen profile, then save.
4. In the browser network tab, view the request/response for the profile `POST`.

**Expected behavior**
The values submitted should have been saved and returned back to the client application. Reloading that item should also reload the data submitted.

**Screenshots**
This screen shows the `POST` request and data sent for a "Profile Specification Profile":
![image](https://user-images.githubusercontent.com/3219480/128736575-a1be78a3-0b7a-48eb-97a8-0ec64764f3c6.png)

This screen shows the response to that `POST` - notice how the `currentValues` are lost:
![image](https://user-images.githubusercontent.com/3219480/128736804-cd59cbbb-829f-4c6e-8ab7-c2ec561c2233.png)

Here is the same issue with a DOI profile:
![image](https://user-images.githubusercontent.com/3219480/128736951-419e616e-e940-41cb-baa4-b8721a44a77b.png)
![image](https://user-images.githubusercontent.com/3219480/128737039-e3e0c7de-5bdc-49c1-85cb-84a106fb24c1.png)

**Comment by olliefreeman** [comment]
This is probably more how the return body is built … as we know the values are saved due to tests, if you perform a “get” the data is in the profile. I don’t think our existing tests check the body of the response after the post

**Comment by pjmonks** [comment]
I tested this in Postman sending the manual profile requests, but could not replicate it. So my original idea of it not working on the backend was false. However, after experimenting with the frontend and reviewing changes on the `ProfileController` recently made, I think I now know what went wrong.

Commit [43010b1c0d6cbdca2c72523d83dc6ce8ff673a30](https://github.com/MauroDataMapper/mdm-core/commit/43010b1c0d6cbdca2c72523d83dc6ce8ff673a30) shows changes made to the `ProfileController`, showing that how the profile instance is bound from the request has changed slightly - by first creating a blank profile instance, then `bindData()` using the request object.

What I found in the UI though was this code:

```typescript
const data = JSON.stringify(result.profile);
return this.resources.profile.saveProfile(
  this.catalogueItem.domainType,
  this.catalogueItem.id,
  selected.namespace,
  selected.name,
  data);
```

The `data` request payload was a JavaScript object but was stringified into a `string` for some reason (the `mdm-resources` function parameter says it should be a `string`, but that seems wrong). My guess is that because a `string` was provided and not a raw JSON object, this interfered with the binding in the `ProfileController`.

By changing the UI to always send a JSON object like this:

```typescript
const data = JSON.parse(JSON.stringify(result.profile));
```

The request is valid and `ProfileController` correctly binds the data submitted.

**Comment by olliefreeman** [comment]
@pjmonks as an FYI the original code in the controller did the same thing as I updated it to do, except it manually worked thru the Json body rather than using the inbuilt grails code to do the same thing. I did however notice that the original code wouldn’t actual map certain things due to an invalid casting. Which is one of the reasons I updated it. Therefore there should have been no difference between the operations before and after. 

This is because both methods essentially turned the Json sent into a jsonObject (map or array)
